samigo/samigo-api/src/java/org/sakaiproject/samigo/util/SamigoConstants.java

```diff
@@ -40,14 +40,60 @@
+    public static final     String      EVENT_ASSESSMENT_SUBMITTED_VIA_URL                  = "sam.assessment.submit.via_url";
+    public static final     String      EVENT_ASSESSMENT_SUBMITTED_THREAD                   = "sam.assessment.thread_submit";
+    public static final     String      EVENT_ASSESSMENT_SUBMITTED_TIMER                    = "sam.assessment.timer_submit";
+    public static final     String      EVENT_ASSESSMENT_SUBMITTED_TIMER_VIA_URL            = "sam.assessment.timer_submit.url";
+
+    //Assessment scoring events
     public static final     String      EVENT_ASSESSMENT_TOTAL_SCORE_UPDATE                 = "sam.total.score.update";
     public static final     String      EVENT_ASSESSMENT_STUDENT_SCORE_UPDATE               = "sam.student.score.update";
     public static final     String      EVENT_ASSESSMENT_AUTO_GRADED                        = "sam.assessmentAutoGraded";
     public static final     String      EVENT_ASSESSMENT_QUESTION_SCORE_UPDATE              = "sam.question.score.update";
 
+    //Assessment manipulation
+    public static final     String      EVENT_ASSESSMENT_SAVEITEM                           = "sam.assessment.saveitem";
+    public static final     String      EVENT_ASSESSMENT_ITEM_DELETE                        = "sam.assessment.item.delete";
+    public static final     String      EVENT_ASSESSMENT_CREATE                             = "sam.assessment.create";
+    public static final     String      EVENT_ASSESSMENT_TAKE                               = "sam.assessment.take";
+    public static final     String      EVENT_ASSESSMENT_TAKE_VIAURL                        = "sam.assessment.take.via_url";
+    public static final     String      EVENT_ASSESSMENT_RESUME                             = "sam.assessment.resume";
+    public static final     String      EVENT_ASSESSMENT_REVIEW                             = "sam.assessment.review";
+    public static final     String      EVENT_ASSESSMENT_REMOVE                             = "sam.assessment.remove";
+    public static final     String      EVENT_ASSESSMENT_REVISE                             = "sam.assessment.revise";
+    public static final     String      EVENT_ASSESSMENT_UNINDEXITEM                        = "sam.assessment.unindexitem";
+    public static final     String      EVENT_ASSESSMENT_PUBLISH                            = "sam.assessment.publish";
+
+    //Published assessment events
+    public static final     String      EVENT_PUBLISHED_ASSESSMENT_REVISE                   = "sam.pubassessment.revise";
+    public static final     String      EVENT_PUBLISHED_ASSESSMENT_SAVEITEM                 = "sam.pubassessment.saveitem";
+    public static final     String      EVENT_PUBLISHED_ASSESSMENT_REMOVE                   = "sam.pubAssessment.remove";
+    public static final     String      EVENT_PUBLISHED_ASSESSMENT_UNPUBLISH                = "sam.pubassessment.confirm_edit";
```

Comment:
Thanks, I'll double check that. I looked at the commit 3-4 times to make sure that the events seemed accurate but feel like one of these might have been missed. Hope it was just my bad naming of a variable. ;)